diff -up katomic-19.08.2/gamewidget.cpp.omv~ katomic-19.08.2/gamewidget.cpp
--- katomic-19.08.2/gamewidget.cpp.omv~	2019-10-14 17:42:45.994398932 +0200
+++ katomic-19.08.2/gamewidget.cpp	2019-10-14 17:43:07.575803067 +0200
@@ -58,7 +58,7 @@ GameWidget::GameWidget ( const QString&
     // TODO uncomment DontSavePainterState optimization back after this bug in Qt 4.6 preview will be
     // fixed (darktears promised it will get fixed in 4.6.0 release). Bug is about wrong coordinates
     // for QPainter passed to QGS::drawForeground() function
-    m_view->setOptimizationFlags( QGraphicsView::DontClipPainter |
+    m_view->setOptimizationFlags( //QGraphicsView::DontClipPainter |
             //QGraphicsView::DontSavePainterState |
             QGraphicsView::DontAdjustForAntialiasing );
 
diff -up katomic-19.08.2/levelset.cpp.omv~ katomic-19.08.2/levelset.cpp
--- katomic-19.08.2/levelset.cpp.omv~	2019-10-14 17:48:29.540739052 +0200
+++ katomic-19.08.2/levelset.cpp	2019-10-14 17:49:00.417127965 +0200
@@ -182,7 +182,7 @@ const LevelData* LevelSet::readLevel(int
 
     for (int j = 0; j < FIELD_SIZE; j++)
     {
-        key.sprintf("feld_%02d", j);
+        key=QString::asprintf("feld_%02d", j);
         QString line = config.readEntry(key,QString());
 
         for (int i = 0; i < FIELD_SIZE; i++)
@@ -234,7 +234,7 @@ const Molecule* LevelSet::readLevelMolec
     int atom_index = 1;
     QString value;
     while (true) {
-        key.sprintf("atom_%c", int2atom(atom_index));
+        key=QString::asprintf("atom_%c", int2atom(atom_index));
         value = config.readEntry(key,QString());
         if (value.isEmpty())
             break;
@@ -259,7 +259,7 @@ const Molecule* LevelSet::readLevelMolec
     int max_i = -1;
     for (int j = 0; j < MOLECULE_SIZE; j++) {
 
-        key.sprintf("mole_%d", j);
+        key=QString::asprintf("mole_%d", j);
         line = config.readEntry(key,QString());
 
         int max_non_null_i = -1;
